src/rt/sections_ldc.d

```diff
@@ -526,3 +522,209 @@ body
 
     return cast(immutable)result;
 }
+
+import core.sys.posix.pthread;
+import core.stdc.config;
+struct tlv_descriptor
+{
+    void* function (tlv_descriptor*) thunk;
+    c_ulong key;
+    c_ulong offset;
+}
+struct segment_command_64
+{
+    uint cmd;
+    uint cmdsize;
+    char[16] segname = 0;
+    ulong vmaddr;
+    ulong vmsize;
+    ulong fileoff;
+    ulong filesize;
+    int maxprot;
+    int initprot;
+    uint nsects;
+    uint flags;
+}
+struct load_command
+{
+    uint cmd;
+    uint cmdsize;
+}
+enum {
+    SECTION_TYPE,
+    LC_SEGMENT_64 = 0x19,
+    S_THREAD_LOCAL_VARIABLES = 0x13
+}
+extern (C) size_t malloc_size(const void* ptr) nothrow @nogc;
+
+/// Represents a TLS range.
+struct TLSRange
+{
+    /// The start of the range.
+    void* start;
+
+    /// The size of the range.
+    size_t size;
+
+    /// Returns `true` if the range is valid.
+    bool isValid() const pure nothrow @nogc @safe
+    {
+        return start !is null && size > 0;
+    }
+
+    /// Returns the range as an array.
+    void[] toArray() pure nothrow @nogc
+    {
+        return start[0 .. size];
+    }
+}
+
+/// Returns the TLS range of the current image.
+TLSRange getTLSRange() nothrow @nogc
+{
+    static ubyte tlsAnchor;
+    const tlsSymbol = &tlsAnchor;
+
+    foreach (i ; 0 .. _dyld_image_count)
+    {
+        const header = cast(const(mach_header_64)*) _dyld_get_image_header(i);
+        auto tlvInfo = tlvInfo(header);
+
+        if (tlvInfo.foundTLSRange(tlsSymbol))
+            return TLSRange(tlvInfo.tlv_addr, tlvInfo.tlv_size);
+    }
+
+    return TLSRange.init;
+}
+
+/**
+ * Returns `true` if the correct TLS range was found.
+ *
+ * If the given `info` is located in the same image as the given `tlsSymbol`
+ * this will return `true`.
+ *
+ * Params:
+ *  info = the TLV info containing the TLV base address
+ *  tlsSymbol = the TLS symbol to search for
+ *
+ * Returns: `true` if the correct TLS range was found
+ */
+bool foundTLSRange(const ref dyld_tlv_info info, const void* tlsSymbol) pure nothrow @nogc
```

Comment:
I think this would be better as a member function of `dyld_tlv_info` named `contains`

Reply:
This is copy-pasted from the DMD2.x patch, so would like to keep it like this. I wouldn't want to differ too much in case changes are made to this code at a later time. What's more, LDC0.x is in maintenance so I think this is fine.

Reply:
Ah, OK.